sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-notification/e-cert-notification.service.ts

```diff
@@ -0,0 +1,157 @@
+  }
+
+  /**
+   * Checks whether the disbursement blocked notification should be created or not.
+   * @param disbursementId disbursement id.
+   * @param entityManager entity manager to execute in transaction.
+   * @returns boolean indicating if the disbursement blocked notification should be created or not.
+   */
+  private async shouldCreateDisbursementBlockedNotification(
+    disbursementId: number,
+    entityManager: EntityManager,
+  ): Promise<boolean> {
+    const notification = await entityManager
+      .getRepository(Notification)
+      .createQueryBuilder("notification")
+      .select("count(*)::int", "notificationCount")
+      .addSelect("max(notification.createdAt)", "maxCreatedAt")
+      .innerJoin("notification.notificationMessage", "notificationMessage")
+      .where("notificationMessage.id = :notificationMessageId", {
+        notificationMessageId:
+          NotificationMessageType.StudentNotificationDisbursementBlocked,
+      })
+      .andWhere("notification.metadata->>'disbursementId' = :disbursementId", {
+        disbursementId,
+      })
+      .getRawOne<NotificationData>();
+    const canSendNewNotification =
+      dateDifference(new Date(), notification.maxCreatedAt) >=
+      BLOCKED_DISBURSEMENT_NOTIFICATION_MIN_DAYS_INTERVAL;
+    // Condition check to create notifications: Less than BLOCKED_DISBURSEMENT_MAXIMUM_NOTIFICATIONS_TO_SEND notifications are created previously and there are no failures in sending those created notifications.
+    // Plus, it has been at least BLOCKED_DISBURSEMENT_NOTIFICATION_MIN_DAYS_INTERVAL days from the last created notification for this disbursement.
+    return (
+      !notification.notificationCount ||
+      (notification.notificationCount <
+        BLOCKED_DISBURSEMENT_MAXIMUM_NOTIFICATIONS_TO_SEND &&
+        canSendNewNotification)
+    );
+  }
+
+  /**
+   * Creates disbursement blocked notifications for student and ministry.
+   * @param disbursementId disbursement id associated with the blocked disbursement.
+   * @param entityManager entity manager to execute in transaction.
+   */
+  private async createDisbursementBlockedNotifications(
+    disbursementId: number,
+    entityManager: EntityManager,
+  ): Promise<void> {
+    const disbursement = await entityManager
+      .getRepository(DisbursementSchedule)
+      .findOne({
+        select: {
+          studentAssessment: {
+            id: true,
+            application: {
+              id: true,
+              applicationNumber: true,
+              student: {
+                id: true,
+                birthDate: true,
+                user: {
+                  id: true,
+                  firstName: true,
+                  lastName: true,
+                  email: true,
+                },
+              },
+            },
+          },
+        },
+        relations: {
+          studentAssessment: { application: { student: { user: true } } },
+        },
+        where: { id: disbursementId },
```

Comment:
Does typeorm not require the ID from the main entity to create the `disbursement` object?

Reply:
Just checked. It returns a whole bunch of extra `properties` without adding the `id` from the base table. Added it. Thanks.